Sources/Cosmos/Crypto/Keys/Keyring.swift

```diff
@@ -4,23 +4,55 @@ import Tendermint
 
 // TODO: This was inside a dependency called 99designs/keyring
 // Check where is the best place to put this
-//
+
+struct KeyringItem: Codable {
+    let key: String
+    let data: Data
+}
+
 // Keyring provides the uniform interface over the underlying backends
 protocol Keyring {
     // Returns an Item matching the key or ErrKeyNotFound
-//    func get(key: string) throws -> Item
+    func get(key: String) throws -> KeyringItem
     // Returns the non-secret parts of an Item
 //    func metadata(key: String) throws -> Metadata
     // Stores an Item on the keyring
-//    func set(item: Item) throws
+    func set(item: KeyringItem) throws
     // Removes the item with matching key
 //    func remove(key: String) throws
     // Provides a slice of all keys stored on the keyring
 //    func keys() throws -> [String]
 }
 
+struct FileKeyring: Keyring {
+    let rootDirectory: String
+   
+    var path: String {
+        rootDirectory + "/file-keyring"
+    }
+
+    init(rootDirectory: String) throws {
+        self.rootDirectory = rootDirectory
+        
+        try FileManager.default.createDirectory(
+            atPath: path,
+            withIntermediateDirectories: true
+        )
+    }
+    
+    func get(key: String) throws -> KeyringItem {
+        let url = URL(fileURLWithPath: path + "/" + key)
+        let data = try Data(contentsOf: url)
+        return KeyringItem(key: key, data: data)
+    }
+     
+    func set(item: KeyringItem) throws {
+        let url = URL(fileURLWithPath: path + "/" + item.key)
+        try item.data.write(to: url)
+    }
+}
```

Comment:
This is the keyring implementation that uses the file system to store the keys as regular files.